hack/ginkgo-e2e.sh

```diff
@@ -183,6 +195,19 @@ case "${E2E_TEST_DEBUG_TOOL:-ginkgo}" in
       program+=("--nodes=25")
     fi
     program+=("${ginkgo_args[@]:+${ginkgo_args[@]}}")
+
+    if [[ "${report_dir:-}" ]]; then
+        # The JUnit report written by the E2E suite gets truncated to avoid
+        # overwhelming the tools that need to process it. For manual analysis
+        # it is useful to have the full reports in both formats that Ginkgo
+        # supports:
+        # - JUnit for comparison with the truncated report.
+        # - JSON because it is a faithful representation of
+        #   all available information.
+        #
+        # This has go be passed to the CLI, the suite doesn't support --output-dir.
+        program+=("--output-dir=${report_dir}" "--junit-report=ginkgo_report.xml" "--json-report=ginkgo_report.json")
```

Comment:
We are passing the option to `ginkgo` (= the CLI), therefore `--junit-report` is the right choice.

But perhaps I misunderstand this comment.

Reply:
I just meant that it's possible to call this script `ginkgo-e2e.sh` with the junit file name specified as `--ginkgo.junit-report=xxx`, now the filename is renamed from `xxx` to `ginkgo_report.xml`.

Reply:
Passing `--ginkgo.junit-report=xxx` still works because it is passed to e2e.test after the `--ginkgo.junit-report` argument that the CLI itself adds. However, then the default `ginkgo_report.xml` is empty because the CLI doesn't get the individual XML files where it expects them. I think that's reasonable because the user gets what they explicitly asked for.